client/middleware/auth.ts

```diff
@@ -4,8 +4,9 @@ export default defineNuxtRouteMiddleware((to, from) => {
   const { isAuthenticated, user } = storeToRefs(useAuth())
 
   if (isAuthenticated.value && user.value!.forceChangePassword) {
-    if (to.fullPath !== '/login?state=change-password') {
-      return navigateTo('/login?state=change-password', { replace: true })
+    if (!to.fullPath.startsWith('/login?state=change-password')) {
```

Comment:
actually, not sure if this is a good idea to check with startWith only, instead of parsing the params. the to object should contain the params, on which we can check if state is change-password